Formula/gap.rb

```diff
@@ -0,0 +1,137 @@
+    #   subdirectories with special names, such as `bin` and `lib`, is not
+    #   suitable for merging with the content of the corresponding
+    #   subdirectories of `/usr/local`.  The easiest temporary solution seems
+    #   to be to drop the compiled bundle into `<prefix>/libexec` and to
+    #   create a symlink `<prefix>/bin/gap` to the startup script.
+    #   This use of `libexec` seems to contradict Linux Filesystem Hierarchy
+    #   Standard, but is recommended in Homebrew's "Formula Cookbook."
+
+    args = %W[--prefix=#{libexec} --with-gmp=system]
+
+    args << "--#{build.with?("readline") ? "with" : "without"}-readline"
+    # NOTE: `--with-readline` is the default, it is included for clarity
+
+    system "./configure", *args
+
+    # Fix a bug caused by the buggy `configure` which does not respect
+    # `--prefix` when generating the startup script: the variable `GAP_DIR`
+    # is being set to the (temporary) build directory, but should be set to
+    # the value of `--prefix` option.
+    ["bin/gap-default32.sh", "bin/gap-default64.sh"].each do |startup_script|
+      next unless File.exist?(startup_script)
+      inreplace startup_script, /^GAP_DIR="[^"]*"$/,
+                                        "GAP_DIR=\"#{libexec}\""
+    end
+
+    system "make"
+
+    libexec.install Dir["*"]
+
+    # Create a symlink `bin/gap` from the symlink `gap.sh`
+    cd libexec/"bin" do
+      # NOTE: the symbolic link `gap.sh` is (or may be) relaive
+      bin.install_symlink File.expand_path(`readlink -n gap.sh`) => "gap"
+    end
+
+    if build.with? "packages"
+      ohai "Trying to automatically build included packages"
+      cd libexec/"pkg" do
+        # NOTE:  running this script is known to produce a number of error
+        #   messages, possibly failing to build certain packages
+        system "../bin/BuildPackages.sh"
+      end
+    end
+  end
+
+  # XXX:  `brew info` displays the caveats according to the options it is
+  #   given, not according to the options with which the formula is installed
+  def caveats
+    if build.without?("packages")
+      <<-EOS.undent
+        If the formula is installed without the `--with-packages' option,
+        some packages in:
+          #{libexec/"pkg"}
+        will need to be built manually with the following script:
+          #{libexec/"bin/BuildPackages.sh"}
+        See the Section 7 of #{libexec/"INSTALL"} for more info.
+      EOS
+    else
+      <<-EOS.undent
+        When the formula is installed with the `--with-packages' option,
+        some packages in
+          #{libexec/"pkg"}
+        are automatically built using the following script:
+          #{libexec/"bin/BuildPackages.sh"}
+        However, this script is known to produce a number of error messages,
+        and thus it might have failed to build certain packages.
+      EOS
+    end
+  end
+
+  test do
+    File.open("test_input.g", "w") do |f|
```

Comment:
Great test!
We prefer the idiom `(testpath/"test_input.g").write <<~EOS`

Reply:
That's actually a pretty skimpy test, only checking the bare minimums.

I have no idea what the standards for your tests are. If they are meant to finished in 1-2 seconds tops, then I guess this is fine. If you can afford a minute (55 seconds on my 2012 MacBook Pro with 2.3 GHz Intel i7 3615QM, i.e. slow by today's standard) , then instead reading the bundled `tst/testinstall.g` provides a much more thorough (and official) way of testing the GAP installation.